### PR TITLE
update imports; remove unused

### PR DESCRIPTION
**Pull request by jbrockmendel**
General cleanup.  flake8 doesn't catch unused cimports.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
> General cleanup. flake8 doesn't catch unused cimports.

can we make that happen?

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -82,12 +79,10 @@ from tslibs.timedeltas cimport cast_from_unit, delta_to_nanoseconds
...
 from tslibs.timezones cimport (
     is_utc, is_tzlocal, is_fixed_offset,
-    treat_tz_as_dateutil, treat_tz_as_pytz,
-    get_timezone, get_utcoffset, maybe_get_tz,
+    treat_tz_as_pytz,
+    get_timezone, maybe_get_tz,
     get_dst_info)
-from tslibs.fields import (
-    get_date_name_field, get_start_end_field, get_date_field,
-    build_field_sarray)
```
Comment (jreback) [inline pandas/_libs/tslib.pyx]: build_field_sarray I think can be completely removed from the codebase

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslib.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx]: It's used in tslibs.resolution.

There are a few funcs scattered around _libs that aren't used anywhere.  Didn't remove them in this PR to keep it uncontroversial.  lib.arrmap and index.get_value_at come to mind.

**Comment by jbrockmendel** [comment]
> can we make that happen?

Not that I'm aware of.  I tried piping the source into flake8 via a sed replacement cimport-->import, but it still chokes when it gets to python syntax errors.

Ultimately I think the effort to build+maintain a flake8 patch to handle this would be more than the effort to just manually check the cimports once in a while.

**Comment by jbrockmendel** [comment]
Would this be easier if broken up into smaller pieces?  I'm hesitant to touch the affected files while this is outstanding.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**